### PR TITLE
kube-controller-manager in a pod.

### PR DESCRIPTION
**Pull request by ArtfulCoder**

### CODE REVIEW

**Comment by erictune** [comment]
Will review when it is ready.

**Comment by ArtfulCoder** [comment]
@dchen1107

**Comment by erictune** [comment]
Is this ready for review?

**Comment by ArtfulCoder** [comment]
Running e2e first..

**Comment by ArtfulCoder** [comment]
Summarizing 1 Failure:

[Fail] Pods [It] should be updated 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:306

Ran 34 of 39 Specs in 912.828 seconds
FAIL! -- 33 Passed | 1 Failed | 1 Pending | 4 Skipped F0417 14:12:46.316009   20085 driver.go:115] At least one test failed
2015/04/17 14:12:46 Error running Ginkgo tests: exit status 255
2015/04/17 14:12:46 Running: teardown

--------------------------------------
Pods 
  should be updated
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:315
[BeforeEach] Pods
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:124
>>> testContext.KubeConfig: /usr/local/google/home/abshah/.kube/.kubeconfig
[It] should be updated
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:315
STEP: creating the pod
STEP: submitting the pod to kubernetes
STEP: waiting up to 5m0s for pod pod-update-e4e68d9f-e545-11e4-ba2a-8851fb5da20e status to be running
INFO: Waiting for pod pod-update-e4e68d9f-e545-11e4-ba2a-8851fb5da20e in namespace default status to be "running" (found "Pending") (40.25983ms)
INFO: Waiting for pod pod-update-e4e68d9f-e545-11e4-ba2a-8851fb5da20e in namespace default status to be "running" (found "Pending") (5.081890826s)
STEP: verifying the pod is in kubernetes
STEP: retrieving the pod
STEP: updating the pod
STEP: deleting the pod

• Failure [10.587 seconds]
Pods
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:630
  should be updated [It]
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:315

  Failed to update pod: pods "pod-update-e4e68d9f-e545-11e4-ba2a-8851fb5da20e" cannot be updated: the resource was updated to 1836

  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:306

**Comment by ArtfulCoder** [comment]
1 test failed. Investigating.

**Comment by ArtfulCoder** [comment]
test passed on re-run.

@erictune Please review

**Comment by derekwaynecarr** [comment]
Before merging can you remove the daemon from cluster/vagrant/util.sh ?  It's inspected as part of verify cluster.  Without that change vagrant hangs each time we move a new thing to a pod

**Comment by ArtfulCoder** [comment]
@derekwaynecarr sure

**Comment by ArtfulCoder** [comment]
PR updated to incorporate Derek's feedback

**Comment by dchen1107** [comment]
cc/ @justinsb on aws support too. Thanks!

**Comment by derekwaynecarr** [comment]
Vagrant LGTM

**Comment by ArtfulCoder** [comment]
@erictune @dchen1107 
PR rebased. unused default file deleted

File: cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+"kind": "Pod",
+"metadata": {"name":"kube-controller-manager"},
+"spec":{
+"hostNetwork": true,
+"containers":[
+    {
+    "name": "kube-controller-manager",
+    "image": "gcr.io/google_containers/kube-controller-manager:{{pillar['kube-controller-manager_docker_tag']}}",
+    "command": [
+                 "/kube-controller-manager",
```
Comment (erictune) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: Seems like `--cluster_name=` is not getting set anymore.  Is that okay?

**Comment by ArtfulCoder** [comment]
@erictune I updated the PR and put in --cluster_name
I verified manually that with my changes, the controller-manager comes up.

**Comment by erictune** [comment]
LGTM

**Pull request merged**